### PR TITLE
feat(behavior_velocity_planner): replace first_stop_path_point_index

### PR DESCRIPTION
**Pull request by yhisaki**
## Description

Previously, we stored the first stopping point on the path in the variable `SceneModuleInterface::first_stop_path_point_index_`. However, we plan to replace this with a class that handles paths continuously in the future by using [autoware_trajectory](https://github.com/autowarefoundation/autoware.universe/tree/main/common/autoware_trajectory). In this case, the current method of directly handling path point indices should be replaced with a method that handles distances from the path's starting point.
In this PR, instead of using the variable `SceneModuleInterface::first_stop_path_point_index_`, we now manage the first_stop_path_point using the variable `SceneModuleInterface::first_stop_path_point_distance_`. `SceneModuleInterface::first_stop_path_point_distance_` represents the distance from the path's starting point.

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: planning/behavior_velocity_planner/autoware_behavior_velocity_detection_area_module/src/scene.cpp [hunk 1]
```diff
@@ -195,13 +195,14 @@ bool DetectionAreaModule::modifyPathVelocity(PathWithLaneId * path, StopReason *
 
   // Create legacy StopReason
   {
-    const auto insert_idx = stop_point->first + 1;
+    double stop_path_point_distance = autoware::motion_utils::calcSignedArcLength(
```
Comment (takayuki5168) [inline planning/behavior_velocity_planner/autoware_behavior_velocity_detection_area_module/src/scene.cpp:198]: const

File: planning/behavior_velocity_planner/autoware_behavior_velocity_no_stopping_area_module/src/scene_no_stopping_area.cpp [hunk 2]
```diff
@@ -153,12 +153,14 @@ bool NoStoppingAreaModule::modifyPathVelocity(PathWithLaneId * path, StopReason
 
     // Create legacy StopReason
     {
-      const auto insert_idx = stop_point->first + 1;
+      double stop_path_point_distance = autoware::motion_utils::calcSignedArcLength(
```
Comment (takayuki5168) [inline planning/behavior_velocity_planner/autoware_behavior_velocity_no_stopping_area_module/src/scene_no_stopping_area.cpp:156]: const

Comment (takayuki5168) [reply planning/behavior_velocity_planner/autoware_behavior_velocity_no_stopping_area_module/src/scene_no_stopping_area.cpp:156]: Please check other variables in this PR.

**Review by takayuki5168 — APPROVED** [review]

**Review by satoshi-ota — APPROVED** [review]
Looks good.

**Review by kyoichi-sugahara — APPROVED** [review]

**Review by kosuke55 — APPROVED** [review]

File: planning/behavior_velocity_planner/autoware_behavior_velocity_traffic_light_module/src/scene.cpp [hunk 3]
```diff
@@ -291,8 +291,11 @@ tier4_planning_msgs::msg::PathWithLaneId TrafficLightModule::insertStopPose(
   // Insert stop pose into path or replace with zero velocity
   size_t insert_index = insert_target_point_idx;
   planning_utils::insertVelocity(modified_path, target_point_with_lane_id, 0.0, insert_index);
-  if (static_cast<int>(target_velocity_point_idx) < first_stop_path_point_index_) {
-    first_stop_path_point_index_ = static_cast<int>(target_velocity_point_idx);
+
+  double target_velocity_point_distance = autoware::motion_utils::calcArcLength(std::vector(
```
Comment (kosuke55) [inline planning/behavior_velocity_planner/autoware_behavior_velocity_traffic_light_module/src/scene.cpp:295]:
```suggestion
  const double target_velocity_point_distance = autoware::motion_utils::calcArcLength(std::vector(
```

Comment (yhisaki) [reply planning/behavior_velocity_planner/autoware_behavior_velocity_traffic_light_module/src/scene.cpp:295]: oh

File: planning/behavior_velocity_planner/autoware_behavior_velocity_planner/src/planner_manager.cpp [hunk 4]
```diff
@@ -103,24 +107,28 @@ tier4_planning_msgs::msg::PathWithLaneId BehaviorVelocityPlannerManager::planPat
...
+      if (firstStopPathPointDistance.value() < first_stop_path_point_distance) {
+        first_stop_path_point_distance = firstStopPathPointDistance.value();
         stop_reason_msg = plugin->getModuleName();
       }
     }
   }
 
-  stop_reason_diag_ = makeStopReasonDiag(
-    stop_reason_msg, output_path_msg.points[first_stop_path_point_index].point.pose);
+  geometry_msgs::msg::Pose stop_pose = autoware::motion_utils::calcInterpolatedPose(
```
Comment (kosuke55) [inline planning/behavior_velocity_planner/autoware_behavior_velocity_planner/src/planner_manager.cpp:128]:
```suggestion
  const geometry_msgs::msg::Pose stop_pose = autoware::motion_utils::calcInterpolatedPose(
```

**Review by rej55 — APPROVED** [review]

**Pull request merged**